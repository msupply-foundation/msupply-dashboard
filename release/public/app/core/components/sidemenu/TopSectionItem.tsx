import React, { FC } from 'react';
import SideMenuDropDown from './SideMenuDropDown';
import { Icon, Link } from '@grafana/ui';
import { NavModelItem } from '@grafana/data';

export interface Props {
  link: NavModelItem;
  onClick?: () => void;
}

const TopSectionItem: FC<Props> = ({ link, onClick }) => {
  const linkContent = (
    <span className="icon-circle sidemenu-icon">
      {link.icon && <Icon name={link.icon as any} size="xl" />}
      {link.img && <img src={link.img} />}
    </span>
  );

  const anchor = link.url ? (
    <Link className="sidemenu-link" href={link.url} target={link.target} onClick={onClick}>
      {linkContent}
    </Link>
  ) : (
    <a className="sidemenu-link" onClick={onClick}>
      {linkContent}
    </a>
  );
  return (
    <div className="sidemenu-item dropdown">
      {anchor}
      <SideMenuDropDown link={link} onHeaderClick={onClick} />
    </div>
  );
};

export default TopSectionItem;
