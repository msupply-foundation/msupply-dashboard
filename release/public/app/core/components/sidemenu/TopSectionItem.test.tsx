import React from 'react';
import { mount } from 'enzyme';
import TopSectionItem from './TopSectionItem';
import { MemoryRouter } from 'react-router-dom';

const setup = (propOverrides?: object) => {
  const props = Object.assign(
    {
      link: {
        text: 'Hello',
        icon: 'cloud',
        url: '/asd',
      },
    },
    propOverrides
  );

  return mount(
    <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
      <TopSectionItem {...props} />
    </MemoryRouter>
  );
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
