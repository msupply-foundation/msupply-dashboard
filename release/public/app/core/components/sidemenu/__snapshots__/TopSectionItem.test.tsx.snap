// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <TopSectionItem
      link={
        Object {
          "icon": "cloud",
          "text": "Hello",
          "url": "/asd",
        }
      }
    >
      <div
        className="sidemenu-item dropdown"
      >
        <Link
          className="sidemenu-link"
          href="/asd"
        >
          <Link
            className="sidemenu-link"
            to="/asd"
          >
            <LinkAnchor
              className="sidemenu-link"
              href="/asd"
              navigate={[Function]}
            >
              <a
                className="sidemenu-link"
                href="/asd"
                onClick={[Function]}
              >
                <span
                  className="icon-circle sidemenu-icon"
                >
                  <Icon
                    name="cloud"
                    size="xl"
                  >
                    <div
                      className="css-1vzus6i-Icon"
                    >
                      <InlineSVG
                        cacheRequests={true}
                        className="css-sr6nr"
                        height={24}
                        src="public/img/iconsunicons/cloud.svg"
                        uniquifyIDs={false}
                        width={24}
                      />
                    </div>
                  </Icon>
                </span>
              </a>
            </LinkAnchor>
          </Link>
        </Link>
        <SideMenuDropDown
          link={
            Object {
              "icon": "cloud",
              "text": "Hello",
              "url": "/asd",
            }
          }
        >
          <ul
            className="dropdown-menu dropdown-menu--sidemenu"
            role="menu"
          >
            <li
              className="side-menu-header"
            >
              <Link
                className="side-menu-header-link"
                href="/asd"
              >
                <Link
                  className="side-menu-header-link"
                  to="/asd"
                >
                  <LinkAnchor
                    className="side-menu-header-link"
                    href="/asd"
                    navigate={[Function]}
                  >
                    <a
                      className="side-menu-header-link"
                      href="/asd"
                      onClick={[Function]}
                    >
                      <span
                        className="sidemenu-item-text"
                      >
                        Hello
                      </span>
                    </a>
                  </LinkAnchor>
                </Link>
              </Link>
            </li>
          </ul>
        </SideMenuDropDown>
      </div>
    </TopSectionItem>
  </Router>
</MemoryRouter>
`;
